graphene_sqlalchemy/converter.py

```diff
@@ -207,97 +214,181 @@ def inner(fn):
 
 def convert_sqlalchemy_column(column_prop, registry, resolver, **field_kwargs):
     column = column_prop.columns[0]
-
+    column_type = getattr(column, "type", None)
+    # The converter expects a type to find the right conversion function.
+    # If we get an instance instead, we need to convert it to a type.
+    # The conversion function will still be able to access the instance via the column argument.
+    if not isinstance(column_type, type):
+        column_type = type(column_type)
     field_kwargs.setdefault(
         "type_",
-        convert_sqlalchemy_type(getattr(column, "type", None), column, registry),
+        convert_sqlalchemy_type(column_type, column=column, registry=registry),
     )
     field_kwargs.setdefault("required", not is_column_nullable(column))
     field_kwargs.setdefault("description", get_column_doc(column))
 
     return graphene.Field(resolver=resolver, **field_kwargs)
 
 
-@singledispatch
-def convert_sqlalchemy_type(type, column, registry=None):
-    raise Exception(
-        "Don't know how to convert the SQLAlchemy field %s (%s)"
-        % (column, column.__class__)
+@singledispatchbymatchfunction
+def convert_sqlalchemy_type(
+    type_arg: Any,
+    column: Optional[Union[MapperProperty, hybrid_property]] = None,
+    registry: Registry = None,
+    **kwargs,
+):
+    registry_ = registry or get_global_registry()
+    # Enable returning Models by getting the corresponding model from the registry
+    # ToDo Move to a separate converter later, matching all sqlalchemy mapped
+    #  types and returning a dynamic
```

Comment:
Yes, I moved it to a function now but will change it to a dynamic later so we can recognize and wrap it in another dynamic in the filter PR - similar to how we do it with relationship filters.

Reply:
Ok I realized we can't use Dynamic here, because graphene can't unpack nested dynamics (e.g. `graphene.List(graphene.Dynamic)`)

So we'll need to check if a field type is a function (`inspect.isfunction(_type) or isinstance(_type, partial)`) additionally to the `instanceof dynamic`check in the filtering PR. This way we still maintain lazy support here.